department-of-veterans-affairs/caseflow
# Ricky/appeals 17558

Resolves #{17558}
[APPEALS-17558](https://vajira.max.gov/browse/APPEALS-17558)

# Description
This update ensures that an attorney, judge or outbound user can remove an MST/PACT selection in the checkbox, when an MST or PACT designation is already assigned. 

This ticket is supposed to pre-populate the MST / PACT checkboxes and retain the checking / unchecking of the box in the redux store.

## Acceptance Criteria

## Testing Plan

1. Go to #{17558} [APPEALS-17558](https://vajira.max.gov/browse/APPEALS-17558)
2. Check out the latest code
3. Log in as Judge (BVAAABSHIRE)
4. Proceed to Queue
5. Select an AMA case
6. Under Actions, activate dropdown and select "Ready for Dispatch"
7. Fill out any form information and for each issue check the following.
  - Is the MST / Pact checkboxes pre-populated correctly from the information tied to the issue?
  - Can the checkbox be checked / unchecked from its starting state?
  - Does the state of the check / unchecked box retain if the modal exits and/or saved?
8. After changing this information, ensure that it is retained by clicking the "edit" link on the appended changes below the issue to see if that information is retained. 

Expected Result:
1. Individual checkboxes for MST / PACT status pre-checked and able to change state.
2. State changes for the checkboxes must be persistent on modal close, save, and form submission.

# Frontend
## User Facing Changes
 - [ ] MST and PACT status are correctly able to be checked / unchecked and have their status viewable in the special issues section.
<img width="1108" alt="Screenshot 2023-05-25 at 2 00 47 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/a41f9d7f-58b4-423a-9116-9cad0ef17e36">
<img width="782" alt="Screenshot 2023-05-25 at 1 55 55 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/adb5ca01-f59c-43f6-a132-685c40b919c6">
<img width="1108" alt="Screenshot 2023-05-25 at 1 56 45 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/39682ad3-09de-4102-ba06-7d1295aa7938">

<img width="1108" alt="Screenshot 2023-05-25 at 2 00 18 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/371115d6-bcde-42d8-a5b8-5fe863e3f83e">
<img width="1108" alt="Screenshot 2023-05-25 at 2 00 23 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/1ab2cdc7-f94c-411f-a5c8-9f8b33420708">

 - [ ] MST and PACT status are reverted back to their original status when the decision is deleted.
<img width="1108" alt="Screenshot 2023-05-25 at 2 00 36 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/8e8674af-9c5e-4f0b-8e9a-181050565565">
<img width="1108" alt="Screenshot 2023-05-25 at 2 00 47 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/5c9d4b3d-3699-451f-8d29-76985b5b6a0d">

 - [ ] MST and PACT status are saved and displayed on case review after decisions have been fully submitted.
<img width="1157" alt="Screenshot 2023-05-25 at 2 03 40 PM" src="https://github.com/department-of-veterans-affairs/caseflow/assets/33578594/32bf52c5-2ca0-4947-b516-e205302c8b69">
